src/poller.ts

```diff
@@ -1,35 +1,134 @@
+import {
+  DEFAULT_INITIAL_CONFIG_REQUEST_RETRIES,
+  DEFAULT_POLL_CONFIG_REQUEST_RETRIES,
+  POLL_JITTER_PCT,
+} from './constants';
+
 export interface IPoller {
   start: () => Promise<void>;
   stop: () => void;
 }
 
 export default function initPoller(
-  interval: number,
+  intervalMs: number,
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   callback: () => Promise<any>,
+  options?: {
+    maxPollRetries?: number;
+    maxStartRetries?: number;
+    errorOnFailedStart?: boolean;
+    pollAfterFailedStart?: boolean;
+  },
 ): IPoller {
   let stopped = false;
+  let failedAttempts = 0;
+  let nextPollMs = intervalMs;
+
+  const start = async () => {
```

Comment:
Because of the different retry strategies and options, it was much easier to move to actually have separate `start()` and `poll()` methods (as opposed to `start = poll`).

Reply:
Yeah, this makes sense to me too